akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -233,6 +233,24 @@ private[http] object StreamUtils {
       def open(): Unit = promise.success(())
     }
   }
+
+  /**
+   * Returns a flow that is almost identity but doesn't pass cancellation from downstream to upstream.
+   *
+   * Note: This might create a resource leak if the upstream never completes. External measures
+   * need to be taken to ensure that the upstream will always complete.
+   */
+  def ignoreCancellation[T]: Flow[T, T, NotUsed] = Flow.fromGraph(new IgnoreCancellation)
+  class IgnoreCancellation[T] extends SimpleLinearGraphStage[T] {
```

Comment:
Yeah the GraphStage upholds all RS rules, so we won't be able to push a closed port - it'll throw "can not push closed port". We could avoid that risk, and make it explicit that this won't happen by doing

```
override def onDownstreamFinish(): Unit = {
  // don't pass cancellation to upstream, and
  setHandler(in, new OutHandler { def onPush(): Unit = log.debug("received unexpected push after cancellation swallowed") } 
}

```